src/product.jl

```diff
@@ -60,9 +60,19 @@ function Base.convert(::Type{ProductSector{T}}, t::Tuple) where {T <: SectorTupl
     return ProductSector{T}(convert(T, t))
 end
 
-Base.one(::Type{ProductSector{T}}) where {I <: Sector, T <: Tuple{I}} = ProductSector((one(I),))
 function Base.one(::Type{ProductSector{T}}) where {I <: Sector, T <: Tuple{I, Vararg{Sector}}}
-    return one(I) ⊠ one(ProductSector{Base.tuple_type_tail(T)})
+    reduce(&, map(MultiFusionStyle, _sectors(T))) == GenericMultiFusion() &&
+        throw(DomainError(ProductSector{T}, "ProductSector $T has multiple units, use `allones` instead of `one`"))
+
+    return only(allones(ProductSector{T}))
+end
+function allones(::Type{ProductSector{T}}) where {I <: Sector, T <: Tuple{I, Vararg{Sector}}}
+    head, rest = Base.tuple_type_head(T), Base.tuple_type_tail(T)
+    if rest == Tuple{}
+        return allones(head)
```

Comment:
This is probably wrong because it unpacks the sectors from the `ProductSector`. In principle, I could have a `ProductSector{Tuple{I}}` with `I` some sector type. You would never encounter this in practical applications, because you would just use `I` directly instead. But in theory it is possible. In that case, `allones` would return a list of `I` objects, whereas you actually want to return a list of `ProductSector{Tuple{I}}` objects.